velox/substrait/VeloxToSubstraitPlan.cpp

```diff
@@ -378,57 +350,87 @@ void VeloxToSubstraitPlanConvertor::toSubstrait(
   aggregateRel->mutable_common()->mutable_direct();
 }
 
-void VeloxToSubstraitPlanConvertor::toSubstrait(
+void VeloxToSubstraitPlanConvertor::toSubstraitJoin(
     google::protobuf::Arena& arena,
-    const std::shared_ptr<const core::AbstractJoinNode>& joinNode,
-    ::substrait::JoinRel* joinRel) {
-  std::vector<core::PlanNodePtr> sources = joinNode->sources();
+    const std::shared_ptr<const core::HashJoinNode> joinNode,
+    ::substrait::Rel* rel) {
+  const auto& joinSources = joinNode->sources();
+  auto joinOutputRowType =
+      joinSources[0]->outputType()->unionWith(joinSources[1]->outputType());
```

Comment:
```
CREATE TABLE table_left(l_a BIGINT, l_b INTEGER, l_c DOUBLE, l_d FLOAT)
CREATE TABLE table_right(r_a BIGINT, r_b INTEGER, r_c DOUBLE, r_d FLOAT)
SELECT l_a FROM table_left JOIN table_right ON l_a = r_a
```
`joinSources[0]->outputType ` will be all columns or just col_a?

Reply:
actually outputType should always reference  left table and/or right table according the join type,  but in velox the HashJoinNode  is actually a combination of  project rel + join rel  due to the output layout can be part of columns in either left and/or right table.